### ISSUE TITLE
Alerting gateways UP and DOWN in TTN V3 (connected versus disconnected)

### ISSUE THREAD

**Issue by marcvanbracht** [TheThingsNetwork/lorawan-stack#4730]
#### Summary
Alerting gateways UP or DOWN in TTN V3 (connected versus disconnected)
Feature: sent an email when status af gateways change (name and current status (from UP tot DOWN of from DOWN to up) + date & time

#### Why do we need this?
The uptime and therefore coverage of the TTN Community network if important. The new TTN V3 stack has a better and more accurate status of a gateway. Know as "Connected" versus "Disconnected"

The status from Connected to Disconnected is shown in the Console seconds/minutes after I bring one of my gateways down.

It should be easy that if you update this status, the Console can sent an email alert about the status change

#### What is already there? What do you see now?
Yes, V3 has an API where everyone can build this feature by them selves. 
1.000 and more users had to program this where it can be achieved from one source; TTN

But I believe it is a key feature of a system that monitors the status of hardware and not informing the user/admin :-(

#### What is missing? What do you want to see?
E-mail alert if the status af o gateway change:
* name of the gateway
* date & time
* status: from UP to DOWN and of course from DOWN to up
* email address (and preferable some more to add ;-) 

#### Environment
The TTN V3 Console where every body logins to

#### How do you propose to implement this?
TTN has already the status of a gateway bij showing in the Console: "Connected"or "Disconnected"
If can't be hard to sent an email when this change as the change is already in TTN V3 Console

#### How do you propose to test this?
I am glas to help. I have 3 gateways at home to test
...

#### Can you do this yourself and submit a Pull Request?
No, I do not have the knowlegde

Best
Marc
www.ttn0478.nl, the 100% open and 100% free LoRaWAN network in NL-Venray

**Comment by kschiffer** [comment]
Thanks Marc for the feature suggestion.

[Cross-posting from the forum](https://www.thethingsnetwork.org/forum/t/ttn-console-v3-and-gateway-monitoring-alerting/52245/7?u=kschiffer):
> Such feature is currently in planning but will still take a while since we will implement it as part of a general Event Server that will allow setting up a whole range of user-defined triggers (e.g. if my-gateway status = disconnected) and actions (e.g. send an email). This will, among other things, support users in a variety of maintenance and monitoring use cases, especially when running large fleets.

We currently don't have a public issue that tracks this, since it hasn't left planning phase yet. I guess we can keep this one open until it can be replaced.

**Comment by htdvisser** [comment]
Adding this to the 2022 Conference milestone, since that's also the milestone for the design part of the Event Server. 

I do want to manage expectations here. As @kschiffer already wrote on the forum, we haven't yet decided if / how much of this is going to be donated to the open-source edition of The Things Stack, and if / how much of this is going to be enabled on The Things Stack Community Edition.

**Comment by htdvisser** [comment]
Re-assigning to @johanstokking and @adriansmares because they're also assigned on https://github.com/TheThingsIndustries/lorawan-stack/issues/1804

**Comment by johanstokking** [comment]
@KrishnaIyer this needs a PM issue.

**Comment by KrishnaIyer** [comment]
Yes this is covered by https://github.com/TheThingsIndustries/product-management/issues/4

**Comment by marcvanbracht** [comment]
Hi Krishna and Johan,

Nice that my request is still on your “to do list”😉

Is there a date set when this feature will be released? I still believe this small feature has a big value for the TTN/TTI network to monitor if gateways are up

Best


Mvg
TTN Venray
Marc van Bracht
Mob: +31 629 022056
KvK: 55969844
www.ttn0478.nl<http://www.ttn0478.nl/>
TTN Venray en TTN0478 is een handelsmerk van 456 Invest BV

***@***.***



Van: Krishna Iyer ***@***.***>
Verzonden: donderdag 29 juni 2023 14:27
Aan: TheThingsNetwork/lorawan-stack ***@***.***>
CC: Marc van Bracht ***@***.***>; Author ***@***.***>
Onderwerp: Re: [TheThingsNetwork/lorawan-stack] Alerting gateways UP and DOWN in TTN V3 (connected versus disconnected) (#4730)


Yes this is covered by TheThingsIndustries/product-management#4<https://github.com/TheThingsIndustries/product-management/issues/4>

**Comment by KrishnaIyer** [comment]
Hey @marcvanbracht: This item is related to one that we track internally. We're tagging this only for issue management purposes. When we discuss the internal issue, we'll consider your proposal.

**Comment by johanstokking** [comment]
@marcvanbracht just for my understanding, do you really want to receive emails for gateways that go down and up? 

Would you have a dedicated email address for that or your personal or work email?

Would you (also or instead) want to receive text messages?

Would it be preferred to (instead) have an integration with Opsgenie, Pagerduty or ServiceNow, which allows you to configure all kinds of on-call and escalation schemes? Because we would not consider that in scope in The Things Stack.

**Comment by marcvanbracht** [comment]
Hi Johan,

For me receiving an email when a gateway goes down and comes back up is all I want.
If this email is only sends to the email address of my TTN account: that is perfect.


Why?
TheThingsNetwork is a community network
TTN is free to use
Therefore I think a plain or simple alerting for gateway DOWN and gateway UP by email is more than enough for me

And:

  *   I can you an email2sms service to receive an sms/text message
  *   I van use an email2webhook service if I want to make more advanced programming

It would be nice to add more email addresses where the alert is sent to of course.
Maybe to all the users with Organisation? I do not know if or how this works as I do not use this section

And if someone wants to now in in-depth details about there gateways and networks…the need to migratie to TTI where it is part of the paid services.

Best and I hope I see this small feature in the next fews months 😉 just before the TTC


Mvg
TTN Venray
Marc van Bracht
Mob: +31 629 022056
KvK: 55969844
www.ttn0478.nl<http://www.ttn0478.nl/>
TTN Venray en TTN0478 is een handelsmerk van 456 Invest BV

***@***.***





Van: Johan Stokking ***@***.***>
Verzonden: vrijdag 30 juni 2023 09:14
Aan: TheThingsNetwork/lorawan-stack ***@***.***>
CC: Marc van Bracht ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [TheThingsNetwork/lorawan-stack] Alerting gateways UP and DOWN in TTN V3 (connected versus disconnected) (#4730)


@marcvanbracht<https://github.com/marcvanbracht> just for my understanding, do you really want to receive emails for gateways that go down and up?

Would you have a dedicated email address for that or your personal or work email?

Would you (also or instead) want to receive text messages?

Would it be preferred to (instead) have an integration with Opsgenie, Pagerduty or ServiceNow, which allows you to configure all kinds of on-call and escalation schemes? Because we would not consider that in scope in The Things Stack.

**Comment by KrishnaIyer** [comment]
This is supported: https://www.thethingsindustries.com/docs/concepts/features/alerting/gateway-connectivity-outages/

**Comment by marcvanbracht** [comment]
Hi,

I see you closed this issue where an alert is sent when a gateways goes offline.

When I am correct this feature is only available for the Things Stack. Is that correct?

Why not make this feature (JUST email alert when gateways goes offline & backup online) for The Things Network?
Why?

  *   You want to have as many gateways UP or not?
  *   Only this part of alerting and NOT sms of webhook for TTN.
  *   I think this should be a basic feature that really helps all of the community members.

With the URL you can query if a gateway is online or not
https://www.thethingsnetwork.org/gateway-data/gateway/ttn0478-gw-011
https://mapper.packetbroker.net/api/v2/gateways/netID=000013%2CtenantID=ttn%2Cid=ttn0478-gw-011
But than everybody needs to make their own | application” to make a alerting system

So please make this feature available for TTN: Only email alert gateway offline & back online

Best regards


Mvg
TTN Venray
Marc van Bracht
Mob: +31 629 022056
KvK: 55969844
www.ttn0478.nl<http://www.ttn0478.nl/>
TTN Venray en TTN0478 is een handelsmerk van 456 Invest BV

***@***.***




Van: Krishna Iyer ***@***.***>
Verzonden: woensdag 30 april 2025 16:48
Aan: TheThingsNetwork/lorawan-stack ***@***.***>
CC: Marc van Bracht ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [TheThingsNetwork/lorawan-stack] Alerting gateways UP and DOWN in TTN V3 (connected versus disconnected) (#4730)

[https://avatars.githubusercontent.com/u/13186113?s=20&v=4]KrishnaIyer left a comment (TheThingsNetwork/lorawan-stack#4730)<https://github.com/TheThingsNetwork/lorawan-stack/issues/4730#issuecomment-2842243716>

This is supported: https://www.thethingsindustries.com/docs/concepts/features/alerting/gateway-connectivity-outages/
